### PR TITLE
linthesia: init at 0.8.0

### PR DESCRIPTION
**Pull request by ckiee**
###### Motivation for this change

Playing piano better!

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/linthesia/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    sqlite
+    SDL2
+    SDL2_ttf
+    SDL2_image
+    gtk3.out # icon cache
+  ];
+
+  meta = with lib; {
+    description = "A game of playing music using a MIDI keyboard following a MIDI file";
+    homepage = "http://linthesia.sourceforge.net";
```
Comment (vcunat) [inline pkgs/games/linthesia/default.nix:34]: Nit: is this homepage really a good choice?  By looking at it I'd think the SW has been abandoned for 10+ years.  I'd suggest `inherit (src.meta) homepage;` instead.

File: pkgs/games/linthesia/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+stdenv.mkDerivation rec {
+  pname = "linthesia";
+  version = "unstable-2022-01-31";
+
+  src = fetchFromGitHub {
+    owner = "linthesia";
+    repo = "linthesia";
+    rev = "11669e355c72fefefb2b3adee31e821dc4c4c9ce";
+    sha256 = "sha256-FmzQjd9rR9NS8za/kgdCACaEqk2J5KpYrjInC0q9Ep4=";
+  };
```
Comment (vcunat) [inline pkgs/games/linthesia/default.nix:13]: There's been a release in the meantime.  Perhaps use that instead?  https://github.com/linthesia/linthesia/releases/tag/0.8.0

**Review by vcunat — COMMENTED** [review]
I'm not sure if it's just me, but rendered text looks like unreadable noise instead.

**Comment by ckiee** [comment]
> I'm not sure if it's just me, but rendered text looks like unreadable noise instead.

A screenshot showing two windows, on the left a terminal emulator running `./result/bin/linthesia` and on the right the linthesia program running, showing a list of folders: favorites, game_themes, hannon, keyboard_classics, learning, popular.

(as for the review comments) Here I was thinking I could get away with a perfect PR this time around, but alas; fixed it now :P


[Now for the obligatory ping in case GitHub decides sending notifications is bad: @vcunat]

**Comment by vcunat** [comment]
It shows OK on _your_ commit, but when I merge that to current master I get
linthesia-noise

**Comment by vcunat** [comment]
I wonder if this log is related to the noise:
```
(process:2388069): GLib-GObject-CRITICAL **: 09:11:42.214: g_object_set_qdata_full: assertion 'quark > 0' failed
```

Anyway, I suggest you try the commit atop current nixpkgs master as well.

**Comment by ckiee** [comment]
> Anyway, I suggest you try the commit atop current nixpkgs master as well.

Rebasing reproduced your screenshot instantly and the culprit appears to be #154827 as reverting that commit fixes it, haven't pushed the revert since I can't reproduce this issue on any other dependants of `SDL2_ttf`.

**Review by vcunat — APPROVED** [review]
OK, assuming it work well for you overall.

**Comment by ckiee** [comment]
A screenshot practically identical to the previous one in this PR

Merge, pretty please?

**Comment by McSinyx** [comment]
Have you contacted upstream about compatibility with SDL_ttf 2.0.18?  Unless it is NixOS-specific problem with its integration, I don't think rushing inclusion of linthesia with SDL_ttf 2.0.15 is the right thing to do here and linthesia should be fixed instead.

**Comment by vcunat** [comment]
I believe it's OK to merge this makeshift solution.

File: pkgs/development/libraries/SDL2_ttf/2.0.15.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    url = "https://www.libsdl.org/projects/SDL_ttf/release/${pname}-${version}.tar.gz";
+    sha256 = "0cyd48dipc0m399qy8s03lci8b0bpiy8xlkvrm2ia7wcv0dfpv59";
+  };
+
+  configureFlags = lib.optional stdenv.isDarwin "--disable-sdltest";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ SDL2 freetype libGL ]
+    ++ lib.optional stdenv.isDarwin darwin.libobjc;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/SDL2_ttf/2.0.15.nix:17]: darwin shouldn't be used as an input like pkgs.

Comment (vcunat) [reply pkgs/development/libraries/SDL2_ttf/2.0.15.nix:17]: I agree, but that's a problem in the `SDL2_ttf` expressions, in both versions.  Not really in this PR.

**Pull request merged**